packages/hl7/src/connection.ts

```diff
@@ -14,14 +14,16 @@ export type Hl7MessageQueueItem = {
   reject?: (err: Error) => void;
 };
 
+export const DEFAULT_ENCODING = 'utf-8';
+
 export class Hl7Connection extends Hl7Base {
   readonly socket: net.Socket;
-  readonly encoding: string;
-  readonly enhancedMode: boolean;
+  encoding: string;
```

Comment:
Removed the `readonly` here so that we can set them later via setters, didn't these fields private though to preserve backwards compat on the public interface